src/main.rs

```diff
@@ -13,18 +17,64 @@ fn main() {
     }
 }
 
-fn run(mut args: impl Iterator<Item = String>) -> Result<()> {
-    while let Some(arg) = args.next() {
-        if arg == "--" {
-            break;
-        }
+
+async fn relay_signals(child: i32) -> Result<()> {
+    // All termination signals that can be caught
+    let mut signals : Signals = Signals::new(&[
+        Signal::Term,
+        Signal::Int,
+        Signal::Quit,
+        Signal::Hup,
+    ])?;
+
+    while let Some(signal) = signals.next().await {
+      let kill = Command::new("kill")
+          .args(["-s", &(signal.unwrap() as i32).to_string(), &child.to_string()])
+          .stderr(Stdio::null())
+          .stdout(Stdio::null())
+          .spawn()?.wait().await;
+      match kill {
+          Ok(_) => { continue; }
+          Err(_) => { break; }
+      }
+
     }
-    let command = args.next().context("comment: no command given")?;
-    let command_args: Vec<String> = args.collect();
-    Command::new(command).args(command_args).spawn()?.wait()?;
     Ok(())
 }
 
+fn run(mut args: impl Iterator<Item = String>) -> Result<()> {
+    tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(async {
+            while let Some(arg) = args.next() {
+                if arg == "--" {
+                    break;
+                }
+            }
+            let command = args.next().context("comment: no command given")?;
+            let command_args: Vec<String> = args.collect();
+            let mut child = Command::new(command).args(command_args).spawn().expect("failed to spawn command");
+            let relayer = relay_signals(child.id().expect("failed to get child PID") as i32);
+            tokio::select! {
+                val = child.wait() => {
+                    match val {
+                        Ok(e) => { exit(e.code().unwrap()) }
+                        Err(e) => {
+                            println!("{}", e);
+                            exit(1)
+                        }
+                    }
+                }
+                _ = relayer => {
+                    println!("Relayer exited")
+                }
+            };
```

Comment:
I wonder if it would be simpler to just spawn a thread for the relayer rather than using tokio? I don't feel strongly at all, but my thinking is that tokio is a pretty big dependency for something as simple as comment.

Reply:
Yeah, I think that makes sense. The downside that made me go this route is that there'd be an extra entry in `top`/`ps` etc

Reply:
I think since we're using the multithreaded tokio runtime that may already be the case, (although I'm not sure when tokio decides to spawn up new threads so maybe not). We could use the single threaded tokio runtime if we want to be sure there aren't any extra threads in the `top` threads view.